packages/rocketchat-lib/server/functions/saveUser.js

```diff
@@ -130,6 +135,10 @@ RocketChat.saveUser = function(userId, userData) {
 			updateUser.$set.name = userData.name;
 		}
 
+		if (userData.role) {
+			updateUser.$set.roles = [ (userData.role || 'user') ];
+		}
+
```

Comment:
@a5his Yes, roles should be an array.

No, you can keep using the database update, I was saying that users should use these methods to add or remove roles, and the `saveUser` method will only override.